zerver/actions/message_edit.py

```diff
@@ -1356,6 +1378,20 @@ def get_message_update_data(
     return content, rendering_result, prior_mention_user_ids, mention_data, links_for_embed
 
 
+def check_resolve_topic_permissions(
+    user_profile: UserProfile,
+    message: Message,
+    stream_id: Optional[int],
+) -> None:
+    if not user_profile.can_resolve_topic():
+        raise JsonableError(_("You don't have permission to resolve/unresolve topics."))
```

Comment:
We don't have any checks for whether it is a stream message or not. I think if we want to allow editing content and `propagate_mode` values other than `change_all` as mentioned in one of the other comments, then we can just reuse `validate_message_edit_payload` function

Reply:
its checked in `is_resolve_topic_request`

Reply:
I cannot see any such check in `is_resolve_topic_request`.

Reply:
updated to reuse `validate_message_edit_payload`